include/dt-bindings/board/am572x-bbai-pins.h

```diff
@@ -106,4 +106,610 @@
+#define P9_22_GPIO_INPUT	P9_22A( PIN_INPUT | MUX_MODE14)							P9_22B( PIN_OUTPUT | MUX_MODE15)	/* xref_clk2.gpio6_19, spi2_sclk.off */		
+#define P9_22_SPI			P9_22B( PIN_OUTPUT_PULLUP | INPUT_EN | MUX_MODE0)		P9_22A( PIN_OUTPUT | MUX_MODE15)	/* spi2_sclk.spi2_sclk, xref_clk2.off */	
+#define P9_22_UART			P9_22B( PIN_OUTPUT_PULLUP | INPUT_EN | MUX_MODE1)		P9_22A( PIN_OUTPUT | MUX_MODE15)	/* spi2_sclk.uart3_rxd, xref_clk2.off */
+
+/* P9_23  (ball A22) gpio7_11 */
+#define P9_23_DEFAULT		P9_23( PIN_OUTPUT_PULLDOWN | INPUT_EN | MUX_MODE14) 	/* spi1_cs1.gpio7_11 */	
+#define P9_23_GPIO			P9_23( PIN_OUTPUT | INPUT_EN | MUX_MODE14) 				/* spi1_cs1.gpio7_11 */		
+#define P9_23_GPIO_PU		P9_23( PIN_OUTPUT_PULLUP | INPUT_EN | MUX_MODE14) 		/* spi1_cs1.gpio7_11 */	
+#define P9_23_GPIO_PD		P9_23( PIN_OUTPUT_PULLDOWN | INPUT_EN | MUX_MODE14) 	/* spi1_cs1.gpio7_11 */	
+#define P9_23_GPIO_INPUT	P9_23( PIN_INPUT | MUX_MODE14) 							/* spi1_cs1.gpio7_11 */
+
+/* P9_24  (ball F20) gpio6_15*/
+#define P9_24_DEFAULT		P9_24( PIN_OUTPUT_PULLUP | INPUT_EN | MUX_MODE14) 		/* gpio6_15.gpio6_15 */	
+#define P9_24_GPIO			P9_24( PIN_OUTPUT | INPUT_EN | MUX_MODE14) 				/* gpio6_15.gpio6_15 */		
+#define P9_24_GPIO_PU		P9_24( PIN_OUTPUT_PULLUP | INPUT_EN | MUX_MODE14) 		/* gpio6_15.gpio6_15 */	
+#define P9_24_GPIO_PD		P9_24( PIN_OUTPUT_PULLDOWN | INPUT_EN | MUX_MODE14) 	/* gpio6_15.gpio6_15 */	
+#define P9_24_GPIO_INPUT	P9_24( PIN_INPUT | MUX_MODE14) 							/* gpio6_15.gpio6_15 */			
+#define P9_24_UART			P9_24( PIN_OUTPUT_PULLUP | INPUT_EN | MUX_MODE3) 		/* gpio6_15.uart10_txd */	
+#define P9_24_CAN			P9_24( PIN_INPUT_PULLUP | MUX_MODE2) 					/* gpio6_15.dcan2_rx  */		
+#define P9_24_I2C			P9_24( PIN_OUTPUT_PULLUP | INPUT_EN | MUX_MODE9) 		/* gpio6_15.i2c3_scl */
+
+/* P9_25  (ball D18) gpio6_17 */
+#define P9_25_DEFAULT		P9_25( PIN_OUTPUT_PULLDOWN | INPUT_EN | MUX_MODE14) 	/* xref_clk0.gpio6_17 */	
+#define P9_25_GPIO			P9_25( PIN_OUTPUT | INPUT_EN | MUX_MODE14) 				/* xref_clk0.gpio6_17 */		
+#define P9_25_GPIO_PU		P9_25( PIN_OUTPUT_PULLUP | INPUT_EN | MUX_MODE14) 		/* xref_clk0.gpio6_17 */	
+#define P9_25_GPIO_PD		P9_25( PIN_OUTPUT_PULLDOWN | INPUT_EN | MUX_MODE14) 	/* xref_clk0.gpio6_17 */	
+#define P9_25_GPIO_INPUT	P9_25( PIN_INPUT | MUX_MODE14) 							/* xref_clk0.gpio6_17 */
+#define P9_25_PRUOUT		P9_25( PIN_OUTPUT_PULLDOWN | INPUT_EN | MUX_MODE13) 	/* xref_clk0.pr2_pru1_gpo5 */
+#define P9_25_PRUIN			P9_25( PIN_INPUT | MUX_MODE12) 							/* xref_clk0.pr2_pru1_gpi5 */
+
+/* P9_26 (ball E21) gpio6_14 & (ball AE2) gpio3_24 */
+#define P9_26_DEFAULT		P9_26A( PIN_OUTPUT_PULLUP | INPUT_EN | MUX_MODE14)		P9_26B( PIN_OUTPUT | MUX_MODE15)	/* gpio6_14.gpio6_14, vin1a_d20.off */	
+#define P9_26_GPIO			P9_26A( PIN_OUTPUT | INPUT_EN | MUX_MODE14)				P9_26B( PIN_OUTPUT | MUX_MODE15)	/* gpio6_14.gpio6_14, vin1a_d20.off */		
+#define P9_26_GPIO_PU		P9_26A( PIN_OUTPUT_PULLUP | INPUT_EN | MUX_MODE14)		P9_26B( PIN_OUTPUT | MUX_MODE15)	/* gpio6_14.gpio6_14, vin1a_d20.off */	
```

Comment:
@jadonk @RobertCNelson I have created these macros to make things a bit cleaner and easier to maintain. These will help in better pin muxing of the BBAI pin headers. Please let me know, what are your thoughts on this?

Reply:
I don't know why this should be easier. I'd think this shouldn't be referenced in multiple places and therefore shouldn't need to be a macro. The mode should become a phandle/symbol in one place and *that* should be referenced, so this macro is totally unnecessary.